cachito/web/content_manifest.py

```diff
@@ -99,8 +99,8 @@ def process_gomod(self, package, dependency, type="icm"):
             if dependency.version and dependency.version.startswith("."):
                 dep_normpath = os.path.normpath(os.path.join(package.name, dependency.version))
                 parent_module_name = gomod.match_parent_module(
-                    dep_normpath, self._gomod_data.keys()
-                )
+                    dependency.name, self._gomod_data.keys()
+                ) or gomod.match_parent_module(dep_normpath, self._gomod_data.keys())
                 # if the dependency is not in this repo, use a relative path from the root module
                 # otherwise just use the parent purl directly
                 if dependency.name not in self._gomod_data:
```

Comment:
I think there's one more issue here with the relative path https://github.com/containerbuildsystem/cachito/blob/5dedcd3f5f14c18877a8924e67c802ef87ec543b/cachito/web/content_manifest.py#L106-L109

If we find the parent module based on `match_parent_module(dependency.name)`
and then try to take the relpath from the parent_module_name to `dep_normpath`
we can get an uncaught exception if the paths are not relative

If we find the parent module by dependency.name, we should take the relative path from parent_module_name to dependency.name

If we find the parent module by dep_normpath, take the relative path to dep_normpath

Reply:
I actually understood that :hushed: